ml/rewrite/templates.py

```diff
@@ -154,3 +171,90 @@ def get_symbols_for_no_confusion_different_constructor_axiom(axiom: kore.Axiom)
             return None
 
         return left.symbol, right.symbol
+
+    # Utils methods for map patterns.
+
+    @staticmethod
+    def is_map_merge_pattern(pattern: kore.Pattern) -> bool:
+        return isinstance(pattern, kore.Application) and \
```

Comment:
we should probably check that the pattern has exactly 2 arguments. otherwise the assertions below doesn't do much